src/EditorFeatures/Core/InheritanceMargin/AbstractInheritanceMarginService_Helpers.cs

```diff
@@ -0,0 +1,178 @@
+            }
+
+            if (memberSymbol is IMethodSymbol or IEventSymbol or IPropertySymbol
+                 && memberSymbol.ContainingSymbol.IsInterfaceType())
+            {
+                return await SymbolFinder.FindMemberImplementationsArrayAsync(
+                    memberSymbol,
+                    solution,
+                    cancellationToken: cancellationToken).ConfigureAwait(false);
+            }
+
+            return ImmutableArray<ISymbol>.Empty;
+        }
+
+        /// <summary>
+        /// Get members overriding the <param name="memberSymbol"/>
+        /// Table for the mapping between images and inheritanceRelationship
+        /// Implemented : I↓
+        /// Implementing : I↑
+        /// Overridden: O↓
+        /// Overriding: O↑
+        /// </summary>
+        private static ImmutableArray<ISymbol> GetOverridingSymbols(ISymbol memberSymbol)
+        {
+            if (memberSymbol is INamedTypeSymbol)
+            {
+                return ImmutableArray<ISymbol>.Empty;
+            }
+            else
+            {
+                using var _ = ArrayBuilder<ISymbol>.GetInstance(out var builder);
+                for (var overridenMember = memberSymbol.GetOverriddenMember();
+                    overridenMember != null;
+                    overridenMember = overridenMember.GetOverriddenMember())
+                {
+                    builder.Add(overridenMember.OriginalDefinition);
+                }
+
+                return builder.ToImmutableArray();
+            }
+        }
+
+        /// <summary>
+        /// Get the derived interfaces and derived classes for <param name="typeSymbol"/>.
+        /// </summary>
+        private static async Task<ImmutableArray<INamedTypeSymbol>> GetDerivedTypesAndImplementationsAsync(
+            Solution solution,
+            INamedTypeSymbol typeSymbol,
+            CancellationToken cancellationToken)
+        {
+            if (typeSymbol.IsInterfaceType())
+            {
+                var allDerivedInterfaces = await SymbolFinder.FindDerivedInterfacesArrayAsync(
+                    typeSymbol,
+                    solution,
+                    transitive: true,
+                    cancellationToken: cancellationToken).ConfigureAwait(false);
+                var allImplementations = await SymbolFinder.FindImplementationsArrayAsync(
+                    typeSymbol,
+                    solution,
+                    transitive: true,
+                    cancellationToken: cancellationToken).ConfigureAwait(false);
+                return allDerivedInterfaces.Concat(allImplementations);
+            }
+            else
+            {
+                return await SymbolFinder.FindDerivedClassesArrayAsync(
+                    typeSymbol,
+                    solution,
+                    transitive: true,
+                    cancellationToken: cancellationToken).ConfigureAwait(false);
+            }
+        }
+    }
```

Comment:
i may have missed it, but where are you finding implementations of interface members?